### ISSUE TITLE
Input params null with XmlSerializer or infinite loop before reaching my service

### ISSUE THREAD

**Issue by dsthode** [DigDes/SoapCore#603]
Hi all,

I'm having quite some trouble using SoapCore in my service. I've developed a minimum example to show my problems hoping someone could point me to where am I doing things wrong. The sample project [is available in this repo](https://github.com/dsthode/soapcore-test).

I've created an WSDL and XSD file with my service and object definitions, then I've created the models in CS with the following command:

`xscgen --namespace "example:soapcore-test=example.soapcore_test" -o Generated soapcore-test.xsd`

and then generated the service stubs with the following command, using `XmlSerializer`:

`dotnet-svcutil soapcore-test.wsdl --sync --serializer xmlserializer --namespace "http://example.com/soapcore-test/,example_com.soapcore_test" --outputdir ServiceStub --projectfile soapcore-test.csproj`

From the service stubs I removed the objects generated and added the `using` statement to use the models generated with `xscgen`.

Then I added a sample service implementation and started the project, setting a breakpoint in my method implementation.

The main highlights from my service creation are the following:
```csharp
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddSoapCore();
            services.AddSingleton(
                typeof(soapcoretest),
                typeof(soapcore_test.Service)
            );
            services.AddMvc();
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            app.UseRouting();
            app.UseEndpoints(endpoints =>
            {
                endpoints.UseSoapEndpoint<soapcoretest>(
                    "/SoapCoreTest",
                    new BasicHttpBinding(),
                    SoapSerializer.XmlSerializer);
            });
        }

```

Then I opened SoapUI and created a project from my WSDL file and generated the sample requests. Then, using that sample request I pressed "Play" and I've had the following results:

1. Using SoapCore version 1.1.0.2, both params to my service implementation came in `null`.
2. Using SoapCore version starting on 1.1.0.4, no request reaches my service, and debugging from a local copy of SoapCore, I found that it's on an infinite loop in `SoapEndpointMiddleware.GetRequestArguments`, where `_serializerHelper.DeserializeInputParameter()`always returns null and the `xmlReader` never advances, so it's stuck on `while (!xmlReader.EOF)`.

Could please someone help me with this issue? I believe this library works by the comments I've read here, and I'm really lost about where to go next and what I'm doing wrong.

Thank you in advance.

**Comment by dsthode** [comment]
I forgot to mention that I'm using .Net Core Runtime 3.1.12 and AspNetCore runtime 3.1.12.

**Comment by dsthode** [comment]
Ok, I'm closing this issue, as I have been able to make more test and have figured out it was a problem with my testing method.

SoapCore, or it's XmlDeserializer, is quite strict with the way the XML message is generated, meaning that if any namespace is missing, it will not deserialize correctly and the values will come null into the method.

And SoapUI failed to identify correctly the namespace for all the XML elements, so after I added a couple of namespaces to the elements here and there the problem was solved and I received the correct values in my method parameters.

Sorry for bothering with non-issues.

Best Regards.
